### DISCUSSION TITLE
copyWith help when using DateTimeColumn get changeTime => dateTime().nullable()();

### DISCUSSION THREAD

**Discussion by matthewkooshad** [simolus3/drift#2475] [Q&A] [answered]
greetings, I just now updated from an old version to the latest, and here's an issue i need help with please.

my stuff table has
DateTimeColumn get changeTime => dateTime().nullable()();

// this complains: The argument type 'DateTime' can't be assigned to the parameter type 'Value<DateTime?>'
stuff = stuff .copyWith(title: title, changeTime: DateTime.now());

how do i resolve this? thanks!

**Comment by simolus3** [comment, marked as answer]
The problem is that, as the value can be nullable, it may not be entirely clear what `copyWith(changeTime: null)` would mean. Should that set the value to `null` or keep the current one? To make things clear, drift wraps nullable fields in a `Value` for `copyWith`. So you could do `changeTime: Value(null)` or `changeTime: Value.absent()` to make things clear.

You can set the `generate_values_in_copy_with` [build option](https://drift.simonbinder.eu/docs/advanced-features/builder_options/) to `false` to revert to the old behavior, or use `changeTime: Value(DateTime.now())`

**Reply by matthewkooshad** [reply]
the intent of that line was to set the changetime to the current datetime.
i changed it to the below, and it doesn't complain now. thanks for the help and the information!
`stuff = stuff.copyWith(changeTime: Value(DateTime.now()));`
